python/triton/runtime/autotuner.py

```diff
@@ -102,14 +103,18 @@ def kernel_call():
             if config.pre_hook:
                 config.pre_hook(full_nargs)
             self.pre_hook(args)
-            self.fn.run(
-                *args,
-                num_warps=config.num_warps,
-                num_stages=config.num_stages,
-                num_ctas=config.num_ctas,
-                **current,
-            )
-            self.post_hook(args)
+            try:
+                self.fn.run(
+                    *args,
+                    num_warps=config.num_warps,
+                    num_stages=config.num_stages,
+                    num_ctas=config.num_ctas,
+                    **current,
+                )
+            except OutOfResources as e:
+                raise e
```

Comment:
My original intention was to inform users whether they encountered the "out of resources" error during the autotuning phase.

How about something like:

```
try:
    self.fn.run(
        *args,
        num_warps=config.num_warps,
        num_stages=config.num_stages,
        num_ctas=config.num_ctas,
        **current,
    )
    exception_raised = False
except Exception as e:
    exception_raised = True
    raise
finally:
    self.post_hook(args, from_exception=exception_raised)
```

Reply:
Sure, except the order of operations here may be a little weird because the `finally` clause will only run after all the other exception handlers, I think?  That's potentially strange because the other exception handlers could literally exit the program.

Maybe you want to run the post_hook *before* the other exception handlers?

```
try:
  self.fn.run()
except as e:
  self.post_hook(args, exception=e)
  raise

self.post_hook(args, exception=None)
```

Reply:
Perhaps even

```
try:
  self.fn.run()
except as e:
  try:
    self.post_hook(args, exception=e)
  finally:
    raise

self.post_hook(args, exception=None)
```

Reply:
> Sure, except the order of operations here may be a little weird because the finally clause will only run after all the other exception handlers, I think? That's potentially strange because the other exception handlers could literally exit the program.

OK, I get it. You were worried that maybe we just crashed the program when compile or run the function. 
I only considered the situation where the compilation will raise an error, which might be shortsighted. 
Thanks for the suggestion.